packages/qwik/src/optimizer/src/plugins/plugin.ts

```diff
@@ -875,8 +875,38 @@ export const manifest = ${JSON.stringify(manifest)};\n`;
   // order by discovery time, so that related segments are more likely to group together
   function manualChunks(id: string, { getModuleInfo }: Rollup.ManualChunkMeta) {
     const module = getModuleInfo(id)!;
-    const segment = module.meta.segment as SegmentAnalysis | undefined;
-    return segment?.entry;
+    const segment = module.meta.segment;
+
+    const optimizer = getOptimizer();
+    const path = optimizer.sys.path;
+    const relativePath = path.relative(optimizer.sys.cwd(), id);
+    const sanitizedPath = relativePath.replace(/^\/+/, '').replace(/\//g, '-');
+
+    if (sanitizedPath.includes('node_modules')) {
+      if (sanitizedPath.includes('core.prod')) {
```

Comment:
why not do a match on `(qwik|core)/dist/core.*js`? And what about qwik-city/router?

Reply:
I think this looks cleaner:

```tsx
    if (id2.endsWith('qwik/dist/core.prod.mjs') || id2.endsWith('qwik/dist/core.prod.cjs')) {
      return 'core';
    }
    if(id2.endsWith('qwik-city/lib/index.qwik.mjs') || id2.endsWith('qwik-city/lib/index.qwik.cjs')) {
      return 'qwik-city';
    }
    ```
    Not sure what the path for qwik/router will be. Probably best to do a separate PR for that.